src/ingest/SpecRepo.cpp

```diff
@@ -295,6 +304,293 @@ void SpecRepo::process(
+            bool isInteger = val.FindMember("Integer") != val.MemberEnd();
+            assert(isFloat || isString || isInteger);
+            if (isInteger) {
+              auto ival = val.FindMember("Integer")->value.FindMember("ival")->value.GetInt();
+              // LOG(INFO) << "rexpr : " << ival;
+              iconfig_val = ival;
+            } else {
+              auto constval = val.FindMember(isFloat ? "Float" : "String")->value.FindMember("str")->value.GetString();
+              //LOG(INFO) << "rexpr : " << constval;
+              config_val = constval;
+              if (isFloat) {
+                fconfig_val = std::stof(constval);
+              }
+            }
+          }
+        }
+        if (isSource) {
+          // hard code
+          table->source = DataSource::KAFKA;
+        }
+        if (isFormat) {
+          table->format = config_val;
+        }
+        if (isLoader) {
+          table->loader = config_val;
+        }
+        if (isMaxMB) {
+          table->max_mb = iconfig_val;
+        }
+        if (isMaxHR) {
+          table->max_seconds = fconfig_val * 3600;
+        }
+        if (isSchema) {
+          table->schema = config_val;
+        }
+        if (isTimestamp) {
+          //hardcode
+          table->timeSpec.type = TimeType::MACRO;
+        }
+        if (isBatch) {
+          auto setting = table->settings;
+          std::ostringstream ss;
+          ss << iconfig_val;
+          std::string s(ss.str());
+          setting.insert({ "batch", s });
+        }
+        if (isProtocol) {
+          table->serde.protocol = config_val;
+        }
+
+        if (isTopicRetention) {
+          table->serde.retention = iconfig_val;
+        }
+
+        if (isTimePartition) {
+          table->timeSpec.pattern = config_val;
+        }
+
+        if (isMacro) {
+          locations = paths;
+        }
+      }
+    }
+  };
+
+  if (table->loader == "ddl") {
+    // TODO: not using yaml
+    const auto createView =
+      "Create view K.pinterest_code (\n"
+      "   service, \n"
+      "   host, \n"
+      "   tag, \n"
+      "   lang, \n"
+      "   stack\n"
+      ") as select dict(service), dict(host), dict(tag), dict(lang), stack\n"
+      "     from kafka where \n"
+      "     format='json' and \n"
+      "     max_mb='400' and\n"
+      "     max_hr=1.3 and\n"
+      "     topic='mytopic' and\n"
+      "     source='s3://nebula/ephemeral/dt={date}/downstream={ds}/contenttype={ct}/pinformat={pf}/eventtype={et}/part-r-{bucket}-fb7ea820-76a3-4c60-be79-956727df7593.gz.parquet' and\n"
+      "     schema='<Row:dict:string,host:string,tag:string,lang:double,stack:string>' and\n"
+      "     loader='Roll' and\n"
+      "     timestamp='provided' and\n"
+      "     batch=500 and\n"
+      "     topic_retention='90000' and\n"
+      "     protocol='binary' and"
+      "     time_partition='DAILY' and"
+      "     macro.et in (1,2,3) and\n"
```

Comment:
@shawncao relates to adding star-partition macro in file path.

Reply:
Thanks Chen, let me take a look tomorrow, feel down already, 🙁

Reply:
refactor with tests running, removed hardcode, should helps with events/dt={date}/action_type={action} use case.

Reply:
In this DDL statement, are you looking to compliant with ANSI SQLL, I haven't found an official definition for `create table definition`, but in my memory of Presto and SQL server, the ANSI SQL style of `create table statement` looks like this: 

```
CREATE TABLE <TBL_NAME>
(
    <col-name> <col-type>,
)
WITH (
   <all table properties> (structure of nested key-values)
)
```

Reply:
BTW, for this DDL language support, what is your goal? why is it better comparing to plain YAML definition?

I'm wondering if we can separate the interfaces into
- keep YAML definition as YAML style
- introduce nebula console (cmd) and allow user to type SQL/DDL statement to interact with Nebula 

Looks like a fun stuff.

Reply:
https://github.com/varchar-io/nebula/issues/62 here is inspiring example I would like to achieve.

Reply:
Yes, that's something we can consider, we need to map what nebula offers to what folks perceive.
- does nebula table support primary key partition
- does  nebula table support setting indexes over arbitrary column(s)

In another view, the #62 is more aligned with materialized view
https://stackoverflow.com/questions/4218657/materialized-view-vs-tables-what-are-the-advantages